### ISSUE TITLE
[Help]: GMTSAR cannot process ALOS2 data aquired in Ultra-fine mode Dual polarization? 

### ISSUE THREAD

**Issue by Twcumtb** [gmtsar/gmtsar#276]
As seen in the title, I used p2p_processing.csh to process ALOS2 data in Ultra-fine mode. But I can not get the interferogram correctly. Can I process this kind of images by GMTSAR?

**Comment by Xiaohua-Eric-Xu** [comment]
I believe so. There might be some parts you need to change manually when it is being interfered with other modes of data. Could you attach a log file?

**Comment by Twcumtb** [comment]
> I believe so. There might be some parts you need to change manually when it is being interfered with other modes of data. Could you attach a log file?

Hi, Xiaohua. I found some "error" during the preprocessing step, here is some massage on the screen ouput.

PREPROCESS - START

Working on images IMG-HH-ALOS2382000544-210619-UBDR1.1__A IMG-HH-ALOS2386140544-210717-UBDR1.1__A ...
rm: No match.
rm: No match.
rm: No match.
pre_proc.csh

 Pre-Process ALOS SLC data - START

 setting SLC_factor to 2.000000 
.... swapping bytes
writing generic LED file: IMG-HH-ALOS2382000544-210619-UBDR1.1__A.LED
 1365081388 integers were clipped 
 258731.501793 median absolute deviation after scaling is 
 ERROR *** reset SCL_factor to something closer to 0.015460 
 setting SLC_factor to 2.000000 
.... swapping bytes
writing generic LED file: IMG-HH-ALOS2386140544-210717-UBDR1.1__A.LED
 1357621013 integers were clipped 
 252250.594271 median absolute deviation after scaling is 
 ERROR *** reset SCL_factor to something closer to 0.015857 
The range sampling rate for master and aligned images are: 104791595

 Pre-Process ALOS SLC data - END

 
PREPROCESS - END


clean up SLC/ folder
rm: No match.

**Comment by dsandwell** [comment]
You need to change the SLC_factor in the config.txt file.

# 2 - start from align SLC images
# SLC scale factor to convert float to int 
SLC_factor = 0.02
#   parameters for focus and align SLC images  #

For this mode it may need to be 0.02 but you have 2.0

The problem is that the floats get converted to int and the small values < 1 get truncated.  This is onl;y an issue with ALOS-½ SLC


> On Feb 19, 2022, at 5:16 PM, Twcumtb ***@***.***> wrote:
> 
> 
> I believe so. There might be some parts you need to change manually when it is being interfered with other modes of data. Could you attach a log file?
> 
> Hi, Xiaohua. I found some "error" during the preprocessing step, here is some massage on the screen ouput.
> 
> PREPROCESS - START
> 
> Working on images IMG-HH-ALOS2382000544-210619-UBDR1.1__A IMG-HH-ALOS2386140544-210717-UBDR1.1__A ...
> rm: No match.
> rm: No match.
> rm: No match.
> pre_proc.csh
> 
> Pre-Process ALOS SLC data - START
> 
> setting SLC_factor to 2.000000
> .... swapping bytes
> writing generic LED file: IMG-HH-ALOS2382000544-210619-UBDR1.1__A.LED
> 1365081388 integers were clipped
> 258731.501793 median absolute deviation after scaling is
> ERROR *** reset SCL_factor to something closer to 0.015460
> setting SLC_factor to 2.000000
> .... swapping bytes
> writing generic LED file: IMG-HH-ALOS2386140544-210717-UBDR1.1__A.LED
> 1357621013 integers were clipped
> 252250.594271 median absolute deviation after scaling is
> ERROR *** reset SCL_factor to something closer to 0.015857
> The range sampling rate for master and aligned images are: 104791595
> 
> Pre-Process ALOS SLC data - END
> 
> PREPROCESS - END
> 
> clean up SLC/ folder
> rm: No match.
> 
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> Triage notifications on the go with GitHub Mobile for iOS or Android. 
> You are receiving this because you are subscribed to this thread.
> 

David T. Sandwell
IGPP 0225
Scripps Inst. of Oceanography
La Jolla, CA
92093-0225

Ph. 858 534-7109
Cell 858 663-9426
http://topex.ucsd.edu

Express Mail Address:
8795 Biological Grade
Room 1102
La Jolla, CA 92037

**Comment by Twcumtb** [comment]
Prof. Sandwell, Thank you very much. It works well with SLC_factor = 0.02. But I see some streaks on the amplitude and correlation figures as well as on the phase images. It seems these streaks are not normal. Is it a problem?

![phasefilt_mask_ll](https://user-images.githubusercontent.com/43894333/154899884-6bc1d92d-54cc-4653-a36c-2c2d84ee4e1b.png)
[display_amp.pdf](https://github.com/gmtsar/gmtsar/files/8106720/display_amp.pdf)
![corr_ll](https://user-images.githubusercontent.com/43894333/154899892-136c43b5-2d78-426d-b372-825ba195d719.png)

**Comment by dsandwell** [comment]
Looks good.

Thess streaks are caused by waves in the ionosphere.  The cause a shift in the pixels.   It may be possible to reduce some of this.

1) look in hte SLC directory.  There is a file called freq_xcorr.dat.  This has the range and azimuth shifts needed to align the two images.
If you plot these shifts having high SNR (column 5) you may see these waves.

2) Look in the processing codes to where xcorr and fitoffset are run.  Make sure fitoffset uses 3 parameters in range and 3 in azimuth.

Otherwise you need to pick a pair with less ionosphere.

> On Feb 20, 2022, at 10:21 PM, Twcumtb ***@***.***> wrote:
> 
> 
> Prof. Sandwell, Thank you very much. It works well with SLC_factor = 0.02. But I see some streaks on the amplitude and correlation figures as well as on the phase images. It seems these streaks are not normal. Is it a problem?
> 
> 
> display_amp.pdf
> 
> 
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> Triage notifications on the go with GitHub Mobile for iOS or Android. 
> You are receiving this because you commented.
> 

David T. Sandwell
IGPP 0225
Scripps Inst. of Oceanography
La Jolla, CA
92093-0225

Ph. 858 534-7109
Cell 858 663-9426
http://topex.ucsd.edu

Express Mail Address:
8795 Biological Grade
Room 1102
La Jolla, CA 92037

**Comment by Xiaohua-Eric-Xu** [comment]
Another way is to do a Multi-Aperture Interferometry (MAI_processing.csh) to see whether there are such stripes.

**Comment by Twcumtb** [comment]
> Another way is to do a Multi-Aperture Interferometry (MAI_processing.csh) to see whether there are such stripes.

Hi, Xiaohua,
    I ran the MAI processing and the script ran smoothly without error messages. But the interferogram looks bad. I can also see stripes on the image. Do I need to change something in the config parameters (confg.alos2.txt) when using MAI_processing.csh? 
![MAI_intf_filt_ll](https://user-images.githubusercontent.com/43894333/155255558-fbf6e611-05f9-4677-848a-7d3867b50ef0.png)
[corr.pdf](https://github.com/gmtsar/gmtsar/files/8121615/corr.pdf)

**Comment by Xiaohua-Eric-Xu** [comment]
This is interesting. No, you don't need to change anything. The MAI is the phase that represent azimuth shift. (Or distortion/mis-registration. Similar as what David suggested for the xcorr results). I do agree this might be ionosphere waves. To further confirm, you could process a different pair and see whether this feature persist (I am assuming not).

**Comment by Twcumtb** [comment]
> This is interesting. No, you don't need to change anything. The MAI is the phase that represent azimuth shift. (Or distortion/mis-registration. Similar as what David suggested for the xcorr results). I do agree this might be ionosphere waves. To further confirm, you could process a different pair and see whether this feature persist (I am assuming not).

 I also ran the interferometric processing with another pair (different acquisition dates for both reference and repeated images) by p2p_processing.csh. The results look good. 

![phasefilt_mask_ll(1)](https://user-images.githubusercontent.com/43894333/155261163-10438ed1-a6f9-436f-bd42-63697e614b03.png)

**Comment by Twcumtb** [comment]
I processed two pairs with a common acquisition date on 2021170 (the two pairs are 2021156-2021170 and 2021170-2021198).  As expected I saw these waves on both the two interferograms and also their correlations. Can I infer that these ionospheric waves are mainly from the common acquisition 2021170?  

![corr_ll(2021156-2021170)](https://user-images.githubusercontent.com/43894333/155279517-aeed5bd8-cbe5-4b55-adeb-713b9a75aefc.png)
![phasefilt_mask_ll(2021156-2021170)](https://user-images.githubusercontent.com/43894333/155279522-c3ee202d-dd91-49fc-8a9e-d31fe1965073.png)
![phasefilt_mask_ll((2021170-2021198))](https://user-images.githubusercontent.com/43894333/155279534-2c346280-e5e5-4a84-85bc-35f2ffea5b68.png)
![corr_ll(2021170-2021198)](https://user-images.githubusercontent.com/43894333/155279544-134b5077-c3b7-40de-8150-87f8287db391.png)
